java/memory/memory-netty/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -1091,6 +1098,40 @@ public void testMemoryLeakWithReservation() throws Exception {
     }
   }
 
+  @Test
+  public void testMemoryUsage() {
+    MemoryLogsAppender memoryLogsAppender = new MemoryLogsAppender();
+    Logger logger = (Logger) LoggerFactory.getLogger("arrow.allocator");
+    logger.setLevel(Level.TRACE);
+    logger.addAppender(memoryLogsAppender);
+    memoryLogsAppender.start();
+
+    try (ArrowBuf buf = new ArrowBuf(ReferenceManager.NO_OP, null,
+        1024, new PooledByteBufAllocatorL().empty.memoryAddress())) {
+      buf.memoryAddress();
+    }
+
+    boolean contains = memoryLogsAppender
+        .contains(
+            Arrays.asList("Memory Usage: \n", "Large buffers outstanding: ", "Normal buffers outstanding: "),
+            Level.TRACE
+        );
+
+    assertTrue(contains);
+  }
+
+  class MemoryLogsAppender extends ListAppender<ILoggingEvent> {
+    public boolean contains(List<String> values, Level level) {
+      return list.stream()
```

Comment:
Just inline this? I don't see why we need a subclass and a method

Reply:
Added